src/components/SettingsSection/RolesSection.tsx

```diff
@@ -0,0 +1,303 @@
+import { z } from 'zod';
+import { usStates } from '@/data/citiesAndStates';
+import { FormRoleEnum, roleAndLegalSchema } from '@/data/formSchemas';
+import { roleOptions } from '@/data/roles';
+import { Box } from '@/styles/containers';
+import { H3 } from '@/styles/text';
+import { Profile, ProfileRole } from '@/types/schema';
+import {
+  capitalize,
+  formatEnumeration,
+  formatTruthy,
+  getCurrentDate,
+  getRoleBools,
+  parseDate,
+  parseDateAlt,
+  parseDateString,
+  timestampStringToDate,
+} from '@/utils/helpers';
+import { useProfileAuth } from '@/utils/hooks';
+import { SettingField, SettingSection } from '.';
+import DateInput from '../DateInput';
+import { FormMessage } from '../Form';
+import InputDropdown from '../InputDropdown';
+import RadioGroup from '../RadioGroup';
+import TextInput from '../TextInput';
+
+const getFormDefaults = (
+  roles: ProfileRole[],
+  profile: Partial<Profile>,
+): Partial<z.infer<typeof roleAndLegalSchema>> => {
+  const roleList = roles.map(role => role.role);
+  const isAttorney = roleList.includes('ATTORNEY');
+  const isLegalFellow = roleList.includes('LEGAL_FELLOW');
+  const isInterpreter = roleList.includes('INTERPRETER');
+
+  let defaultRole: FormRoleEnum = '';
+  if (isInterpreter) defaultRole = 'INTERPRETER';
+  if (isAttorney) defaultRole = 'ATTORNEY';
+  if (isLegalFellow) defaultRole = 'LEGAL_FELLOW';
+  if (isAttorney && isInterpreter) defaultRole = 'ATTORNEY,INTERPRETER';
+  if (isLegalFellow && isInterpreter) defaultRole = 'LEGAL_FELLOW,INTERPRETER';
+
+  const defaultValue: Partial<z.infer<typeof roleAndLegalSchema>> = {
+    barNumber: profile.bar_number,
+    eoirRegistered: profile.eoir_registered,
+    expectedBarDate: profile.expected_bar_date
+      ? new Date(profile.expected_bar_date)
+      : undefined,
+    stateBarred: profile.state_barred,
+    roles: defaultRole,
+  };
+
+  return defaultValue;
+};
+
+const getDateDefault = (profile: Partial<Profile>): string =>
+  profile.expected_bar_date
+    ? parseDateAlt(timestampStringToDate(profile.expected_bar_date))
+    : '';
+
+export default function RolesSection() {
+  const { profile, auth } = useProfileAuth();
+  const [isEditing, setIsEditing] = useState(false);
+
+  const [expectedBarDate, setExpectedBarDate] = useState<string>(
+    getDateDefault(profile.profileData ?? {}),
+  );
+
+  const form = useForm<z.infer<typeof roleAndLegalSchema>>({
+    resolver: zodResolver(roleAndLegalSchema),
+    defaultValues: getFormDefaults(profile.roles, profile.profileData ?? {}),
+  });
+
+  const onValidSubmit = async (values: z.infer<typeof roleAndLegalSchema>) => {
+    const { userId } = auth;
+    if (!userId) throw new Error('User ID not found! Are you logged in?');
+
+    // parse roles to update into array
+    // nasty boolean checking to ensure type safety
+    const rolesToUpdate: ProfileRole[] = [];
+
+    const { isAttorney, isLegalFellow, isInterpreter } = getRoleBools(
```

Comment:
is this helper necessary here? This seems to also be the only be the only place the helper is used.
Can't we just do the check in the lines below? 

`if (values.roles.includes('ATTORNEY'))`, etc.

Reply:
yeah lowkey thought id need it in more places
i can remove it lol